build.sh

```diff
@@ -18,17 +18,23 @@ PRINT_USAGE() {
     echo "build.sh [options]"
     echo ""
     echo "options:"
-    echo "--help, -h : Show help"
+    echo "-j          : Multicore build (i.e. -j:3 for 3 cores)"
+    echo "-icu        : Path to ICU include folder (see example below)"
+    echo "--help, -h  : Show help"
     echo "--debug, -d : Debug build"
     echo "CXX=<path to clang++>"
     echo "CC=<path to clang>"
     echo ""
-    echo "example: ./build.sh CXX=/path/to/clang++ CC=/path/to/clang"
+    echo "example : ./build.sh CXX=/path/to/clang++ CC=/path/to/clang -j:2"
+    echo "with icu: ./build.sh -icu:/usr/local/Cellar/icu4c/version/include/"
+    echo ""
 }
 
 _CXX=""
 _CC=""
 DEBUG_BUILD=0
+MULTICORE_BUILD=""
```

Comment:
nproc is not cross platform.

Apart from dealing with cross platform nproc.. 

Having multicore mode enabled; I have failed to see my system doesn't have uchar.h. Enabling multicore by default is no good for development and this work is still under development.